composer/trainer/trainer.py

```diff
@@ -1251,6 +1334,7 @@ def __init__(
         if parallelism_config is not None:
             # Patch PyTorch to fix distributed bugs
             patch_pytorch()
+            patch_unshard_for_automicrobatching(False)
```

Comment:
Can we only patch if auto_microbatching is True?

Reply:
good catch